pkg/proc/variables.go

```diff
@@ -438,8 +438,15 @@ func getGVariable(thread Thread) (*Variable, error) {
 
 	gaddr, hasgaddr := regs.GAddr()
 	if !hasgaddr {
-		var err error
-		gaddr, err = readUintRaw(thread.ProcessMemory(), regs.TLS()+thread.BinInfo().GStructOffset(), int64(thread.BinInfo().Arch.PtrSize()))
+		offset := thread.BinInfo().GStructOffset()
```

Comment:
move this logic inside (*BinaryInfo).GStructOffset. Also instead of having a derefGStructOffset just check Arch.Name and BinaryInfo.GOOS.

Reply:
> move this logic inside (*BinaryInfo).GStructOffset. 

If I understood you correctly, to do this I would have to update `GStructOffset` to accept a `MemoryReader` and return an error. Am I right?

Reply:
Yes, that would be fine.

Reply:
Done!